src/multi.c

```diff
@@ -105,7 +105,13 @@ void discardTransaction(client *c) {
 /* Flag the transaction as DIRTY_EXEC so that EXEC will fail.
  * Should be called every time there is an error while queueing a command. */
 void flagTransaction(client *c) {
-    if (c->flag.multi) c->flag.dirty_exec = 1;
+    if (c->flag.multi) {
+        c->flag.dirty_exec = 1;
+        if (c->mstate.commands) {
```

Comment:
freeClientMultiState seems safe to call without the if (commands) check.

Reply:
free will do nothing if the commands is NULL (or the count is 0), and init is actually  doesn't do anything complicated. 

i am ok with it, just bother with the if block, maybe we can mix it with a new freeAndInitxxx?

Reply:
There is no difference between two lines of code and one line of code. If you insist that this is better, then okay.

Reply:
I don't insist. :)